File: backend/internal/domain/option.go
package domain

import "time"

// OptionState – состояние временной опции относительно постоянной сущности.
type OptionState string

const (
	OptionStateActual  OptionState = "ACTUAL"
	OptionStateNew     OptionState = "NEW"
	OptionStateChanged OptionState = "CHANGED"
	OptionStateDeleted OptionState = "DELETED"
)

// Option представляет опцию вопроса, которая хранится в таблице survey_options
type Option struct {
	ID          int       `json:"id" db:"id"`
	QuestionID  int       `json:"question_id" db:"question_id"`
	Label       string    `json:"label" db:"label"`
	OptionOrder int       `json:"option_order" db:"option_order"`
	CreatedAt   time.Time `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time `json:"updated_at" db:"updated_at"`
}

// SurveyOptionsTemp - временная таблица для опций.
type OptionTemp struct {
	ID               int         `json:"id" db:"id"`
	OptionOriginalID *int        `json:"option_original_id" db:"option_original_id"`
	QuestionID       int         `json:"question_id" db:"question_id"`
	Label            string      `json:"label" db:"label"`
	OptionOrder      int         `json:"option_order" db:"option_order"`
	OptionState      OptionState `json:"option_state" db:"option_state"`
	CreatedAt        time.Time   `json:"created_at" db:"created_at"`
	UpdatedAt        time.Time   `json:"updated_at" db:"updated_at"`
}

File: frontend/survey-frontend/src/types/option.ts
// src/types/option.ts

/** Временные и реальные состояния опции */
export enum OptionState {
  Actual  = "ACTUAL",
  New     = "NEW",
  Changed = "CHANGED",
  Deleted = "DELETED",
}

/** Базовая структура опции, возвращаемая API */
export interface Option {
  id: number;
  question_id: number;
  label: string;
  option_order: number;
  option_state: OptionState;
  created_at: string; // ISO
  updated_at: string; // ISO
}

/** Ответ API при работе с одной опцией */
export interface OptionResponse {
  option: Option;
}

/** Пакет всех опций для вопроса */
export interface OptionsListResponse {
  options: Option[];
}

/** Тело POST /option */
// eslint-disable-next-line @typescript-eslint/no-empty-object-type
export interface CreateOptionRequest {
  // no body: type inferred from question
}

/** Тело PATCH /order */
export interface UpdateOptionOrderRequest {
  new_order: number;
}

/** Тело PATCH / (label) */
export interface UpdateOptionLabelRequest {
  label: string;
}
